src/include/souffle/datastructure/EquivalenceRelation.h

```diff
@@ -172,19 +172,21 @@ class EquivalenceRelation {
         assert(toInsert.size() == (std::size_t)size);
 
         // add the intersecting dj sets into this one
-        {
-            value_type el;
-            value_type rep;
-            auto it = other.sds.sparseToDenseMap.begin();
-            auto end = other.sds.sparseToDenseMap.end();
-            for (; it != end; ++it) {
-                std::tie(el, std::ignore) = *it;
-                rep = other.sds.findNode(el);
+        other.genAllDisjointSetLists();
+        for (typename StatesMap::chunk it : other.equivalencePartition.getChunks(MAX_THREADS)) {
```

Comment:
I'm a bit confused by this: the mention of `MAX_THREADS` here (as copied from `insertAll`) seems to indicate that there's some kind of parallelism happening, but it's not like this for-loop starts with `#pragma openmp parallel for` or anything obvious like that... I would appreciate more seasoned C++ programmers thoughts here:

1. Is any parallelism happening here/in `insertAll`? If so, how can you tell?
2. If not, is there any reason to iterate over this in chunks? Should/can I just use the (non-chunked) iterator over `btree`?

Reply:
Looking at the implementation of `getChunks`, it looks like it does exactly what it sounds like: It iterates over the set, and splits it into `n` sequences of length approx. `|C| / n`.

Looks like a copy-paste mistake. I'd say remove it entirely, both here and in `insertAll`, and just iterate over the collection.